torch/fx/utils.py

```diff
@@ -0,0 +1,144 @@
+import torch
+import torch.fx
+import copy
+
+def extract_module(mod : torch.fx.GraphModule, target_qualname : str) -> torch.fx.GraphModule:
+    """
+    Given a GraphModule and a qualified name for a Module resident in the
+    GraphModule's module hierarchy, return a new GraphModule that is equivalent
+    to the original, but with the nodes originating from `target_qualname`
+    out-lined (opposite of inlined).
+    """
+    target_module : torch.nn.Module = mod
+    for atom in target_qualname.split('.'):
+        target_module = getattr(target_module, atom)
+
+
+    # ===== Stage 1: identify Nodes to split off and external data deps =====
+    block_nodes = []
+
+    # Keep track of data dependencies used and referenced in the
+    # extracted block and the rest of the code, respectively.
+    #
+    # We'll use this later to figure out what needs to be inputs/outputs
+    # to/from the extracted block
+    block_defs = set()
+    block_uses = set()
+    rest_defs = set()
+    rest_uses = set()
+
+    for node in mod.graph.nodes:
+        def in_block(node):
+            return node.module_qualname and node.module_qualname.startswith(target_qualname)
```

Comment:
nit: `startswith` can misbehave as `root.foobar` starts with `root.foo` :) we probably need a helper function which would do the right thing (or store the qual name as already split by `.`)

Reply:
Yeah, good point! Will fix